Source/Core/Core/AchievementManager.cpp

```diff
@@ -246,6 +252,50 @@ void AchievementManager::DoFrame()
   }
 }
 
+bool AchievementManager::CanPause()
+{
+  u32 frames_to_next_pause = 0;
+  bool can_pause = rc_client_can_pause(m_client, &frames_to_next_pause);
+  if (!can_pause)
+  {
+    OSD::AddMessage("Cannot spam pausing in hardcore mode.", OSD::Duration::VERY_LONG,
+                    OSD::Color::RED);
+    OSD::AddMessage(
+        fmt::format("Can pause in {} seconds.", static_cast<float>(frames_to_next_pause) / 60),
```

Comment:
This shouldn't be using a hardcoded 60 (unless RetroAchievements has a hardcoded behavior of considering 16.67 ms to be one frame, which I really hope it doesn't). You can get the actual refresh rate from GetTargetRefreshRate in VideoInterface.h. You may also want to adjust for the emulation speed, though I'm not how sure how useful that is since you can start and stop fast forwarding very quickly, so I'll leave the choice on that one to you.

(Actually, how does rcheevos know the current framerate? Is it just not taking the framerate into account for determining when you can pause, making the pause interval longer for 50 Hz games in terms of seconds?)

Reply:
It assumes 60Hz. No, the logic behind it doesn't make much sense to me either.